Lesson6/HomeWork.py

```diff
@@ -0,0 +1,43 @@
+"""
+HomeWork after lesson 6.
+
+we have two lists with equal or different size
+ex. l1=[1,3,5,7]  l2=[1,4,5]
+
+task:
+create list that will store such values
+list_target = [(1,1), (3,4), (5,5), (7,0)]
+zero (0) is our default value that we set if no
+such element by index was found in certain list.
+code should work and vise versa
+ex. l1=[1,4,5] l2=[1,3,5,7] input data should
+produce list_target = [(1,1), (4,3), (5,5), (0,7)]
+your solution should include comprehension constructions
+
+Advices:
+set of (list1 indexes union list2 indexes) could
+be helpful to get larger indexes scope ( or use if-else)
+dict as you remember has default value if key was
+not found d1.get(key, 0)
+"""
+
+l1 = [2, 4, 6, 8, 10]
+l2 = [1, 2, 3]
+
+"""
+Here is the solution:
+1) create dict with keys from l1 and values from l2
+2) get longer dict length for next comprehension
+3) form final result as values from dicts
+"""
+
+# Step 1
+dict_a = dict(enumerate(l1, start=0))
+dict_b = dict(enumerate(l2, start=0))
+
+# Step 2
+dict_len = len(dict_a) if len(dict_a) >= len(dict_b) else len(dict_b)
```

Comment:
If you already have dicts with indexes - This could be a little bit simplified ( just read Advice section)  ^^ )
Smth aka
```
default = 0
d1 = {key: val for key, val in enumerate(l1)}
d2 = {key: val for key, val in enumerate(l2)}
l3 = [(d1.get(key, default), d2.get(key, default)) for key in set(d1.keys() | d2.keys())]  
```